scserver.js

```diff
@@ -103,7 +103,27 @@ var SCServer = function (options) {
     EventEmitter.prototype.emit.call(self, 'ready');
   });
 
-  var wsEngine = require(opts.wsEngine);
+  // Provide explicit strings in order to make dynamic `require` compatible with Webpack.
+  var wsEngine = undefined;
+  try {
+    switch(opts.wsEngine) {
+      case('ws'): {
+        wsEngine = require('ws');
+        break;
+      }
+      case('uws'): {
+        wsEngine = require('uws');
+        break;
+      }
+      default: {
+        // This won't work in Webpack, but will work elsewhere.
+        wsEngine = require(opts.wsEngine);
+        break;
+      }
+    }
+  } catch(e) {
+    if(!e instanceof InvalidOptionsError) throw new InvalidOptionsError('wsEngine is not installed.')
+  }
```

Comment:
Thanks for raising this issue.
It's a tricky one so here is some feedback:

- This try-catch block could be bad 'wsEngine is not installed.' is not the only possible error that could happen here (assuming that the user provides a custom wsEngine); I think it's better to not catch the error from the module here and just let it propagate (like before)
- SC no longer uses `uws` (because it's been deprecated), instead it uses SC's own fork which is called `sc-uws`.
- The code style is inconsistent with the rest of the code. Notice that there are spaces; `if (...)` not `if(...)` (same with switch, case and catch statements). Also you can declare `var wsEngine;` instead of `var wsEngine = undefined;`.

But I think maybe the best solution would be to allow `wsEngine` to be either a string or an object. If it's a string, we require it dynamically, otherwise we assume that the user has passed the module directly and so just assign the object directly to the `wsEngine` property. That way WebPack users can still use socketcluster-server with a custom WS engine.

Reply:
That said, if you like the convenience of passing a string and want that supported for WebPack, then feel free to do use the switch statement approach.

Reply:
Thanks for feedback!
1. Makes sense, let's let the error propogate.
2. No problem, I will restyle it.

Good idea, in fact passing wsEngine is even more convenient (and less webpack warnings).